### PR TITLE
store/mockstore/unistore: migrate unistore core lib code to tidb repo for removing tidb dependency on unistore

### PR DESCRIPTION
**Pull request by hslam**
### What problem does this PR solve?

Problem Summary: tidb and unistore are interdependent.

### What is changed and how it works?

Migrate unistore core lib code to tidb repo for removing tidb dependency on unistore.

### Related changes

- Nah

### Check List 

Tests 

- Unit test

Side effects

- Nah

### Release note 

- No longer rely on unistore module.

### CODE REVIEW

File: store/mockstore/unistore/tikv/mvcc/mvcc.go [hunk 1]
```diff
@@ -98,6 +98,7 @@ func (l *MvccLock) MarshalBinary() []byte {
 	}
 	copy(buf[cursor:], l.Value)
 	cursor += len(l.Value)
+	_ = cursor
```
Comment (disksing) [inline store/mockstore/unistore/tikv/mvcc/mvcc.go:99]: can we remove the 2 lines?

File: store/mockstore/unistore/tikv/mvcc_test.go [hunk 2]
```diff
@@ -965,6 +965,7 @@ func (s *testMvccSuite) TestMvccTxnRead(c *C) {
 	MustGetNone(k1, 1, store)
 
 	v1 = []byte("v1lock")
+	_ = v1
```
Comment (disksing) [inline store/mockstore/unistore/tikv/mvcc_test.go:976]: can we remove `v1`.

File: store/mockstore/unistore/tikv/mvcc/mvcc.go (hunk 1 above)
Comment (hslam) [reply store/mockstore/unistore/tikv/mvcc/mvcc.go]: Removed.

File: store/mockstore/unistore/tikv/mvcc_test.go (hunk 2 above)
Comment (hslam) [reply store/mockstore/unistore/tikv/mvcc_test.go]: Removed.

**Review by disksing — APPROVED** [review]

**Pull request merged**